### PR TITLE
sqlalchemy-execute-raw-query: fix rule to take params into consideration

### PR DESCRIPTION
**Pull request by 0xdbe**

### CODE REVIEW

File: python/sqlalchemy/security/sqlalchemy-execute-raw-query.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 engine = create_engine('postgresql://user@localhost/database')
 echo("database connexion: ok")
 # ruleid: sqlalchemy-execute-raw-query
-engine.execute("INSERT INTO person (name) VALUES ('" + name + "')")
+engine.execute("INSERT INTO person (name) VALUES ('" + name + "')", multi=False)
```
Comment (mschwager) [inline python/sqlalchemy/security/sqlalchemy-execute-raw-query.py:11]: Can you make this a completely new `# ruleid: ...` test? That way we have both tests confirming behavior :)

File: python/sqlalchemy/security/sqlalchemy-execute-raw-query.py [hunk 2]
```diff
@@ -193,3 +193,34 @@
 engine = create_engine('postgresql://user@localhost/database')
 query = select(literal_column("users.fullname", String) + ', ' + literal_column("addresses.email_address").label("title")).where(and_(literal_column("users.id") == literal_column("addresses.user_id"), text("users.name BETWEEN 'm' AND 'z'"), text("(addresses.email_address LIKE :x OR addresses.email_address LIKE :y)"))).select_from(table('users')).select_from(table('addresses'))
 conn.execute(query, {"x":"%@aol.com", "y":"%@msn.com"}).fetchall()
+
+
+# SQL Composition using SQL Expression
+# ok: sqlalchemy-execute-raw-query
```
Comment (mschwager) [inline python/sqlalchemy/security/sqlalchemy-execute-raw-query.py]: Unless I'm missing something, I think you want these `ok` annotations right above the `execute` line :+1:

File: python/sqlalchemy/security/sqlalchemy-execute-raw-query.py [hunk 3]
```diff
@@ -193,3 +193,34 @@
...
+  meta.reflect(bind=connection)
+  product_table = meta.tables['product']
+  stmt = (
+    select(product_table)
+    .where(product_table.columns[field_name] == value_name)
+  )
+  result = connection.execute(stmt)
+
+# Insert multi data record using SQL Expression
+# ok: sqlalchemy-execute-raw-query
```
Comment (mschwager) [inline python/sqlalchemy/security/sqlalchemy-execute-raw-query.py]: Same as above.

File: python/sqlalchemy/security/sqlalchemy-execute-raw-query.yaml [hunk 4]
```diff
@@ -7,7 +7,8 @@ rules:
        safely, prepared statement should be used.
        SQLAlchemy provides TextualSQL to easily used prepared statement with
        named parameters. For complexe SQL composition, use SQL Expression
-       Langague or SQL Statements and Expressions API.
+       Langague or Schema Definition Language. In most case, SQLAlchemy ORM will
```
Comment (mschwager) [inline python/sqlalchemy/security/sqlalchemy-execute-raw-query.yaml]:
```suggestion
       Language or Schema Definition Language. In most cases, SQLAlchemy ORM will
```

**Review by mschwager — COMMENTED** [review]
Nice catch! Thank you!

File: python/sqlalchemy/security/sqlalchemy-execute-raw-query.py (hunk 2 above)
Comment (0xdbe) [reply python/sqlalchemy/security/sqlalchemy-execute-raw-query.py]: I have to check if concatenation is use before execution:

```
# Insert multi data record using SQL Expression
connection_string = 'sqlite:///db.sqlite'
engine = create_engine(connection_string, echo=True)
with engine.connect() as connection:
  meta = MetaData()
  meta.reflect(bind=connection)
  product_table = meta.tables['product']
  # ruleid: sqlalchemy-execute-raw-query
  stmt = insert(product_table) + 'test'
  values = [
      {field_name: 'hazelnut', field_price: 5},
      {field_name: 'banana', field_price: 8}
  ]
  connection.execute(stmt, values)`
```

`ok` annotations should be just before `stmt` line ?

**Pull request closed without merging**